chebi-update/src/main/java/org/reactome/release/chebiupdate/ChebiUpdater.java

```diff
@@ -114,196 +152,263 @@ public void updateChebiReferenceMolecules() throws SQLException, Exception
+		{
+			for (String message : this.referenceEntityChanges.get(creator))
+			{
+				refEntChangeLog.info("{}\t{}",creator.toString(), message);
+			}
+		}
 	}
 
 	/**
-	 * Update a molecule's formula.
+	 * Update a molecule's formula. Should always update (set to the ChEBI formula) if the current formula and the formula from ChEBI differ.
 	 * 
 	 * @param molecule
 	 * @param chebiFormulae
-	 * @param moleculeFormulae
-	 * @param prefix
+	 * @return true if the formula was updated, or if the field was populated (i.e. was NULL before). False, otherwise.
 	 * @throws InvalidAttributeException
 	 * @throws InvalidAttributeValueException
 	 * @throws Exception
 	 */
-	private void updateMoleculeFormula(GKInstance molecule, List<DataItem> chebiFormulae, String moleculeFormulae, String prefix) throws InvalidAttributeException, InvalidAttributeValueException, Exception
+	private boolean updateMoleculeFormula(GKInstance molecule, List<DataItem> chebiFormulae) throws InvalidAttributeException, InvalidAttributeValueException, Exception
 	{
+		String moleculeFormulae = (String) molecule.getAttributeValue(ReactomeJavaConstants.formula);
+		boolean updated = false;
+		String moleculeIdentifier = (String) molecule.getAttributeValue(ReactomeJavaConstants.identifier);
+		String reportLinePrefix = "ReferenceMolecule (DB ID: " + molecule.getDBID() + " / ChEBI ID: " + moleculeIdentifier + ") has changes: ";
 		if (!chebiFormulae.isEmpty())
 		{
 			String firstFormula = chebiFormulae.get(0).getData();
-			if (firstFormula != null)
+			if (firstFormula != null && !firstFormula.trim().equals(""))
 			{
 				if (moleculeFormulae == null)
 				{
-					this.formulaFillSB.append(prefix).append("New Formula: ").append(firstFormula).append("\n");
+					this.formulaFillSB.append(reportLinePrefix).append("New Formula: ").append(firstFormula).append("\n");
+					updated = true;
 				}
 				else if (!firstFormula.equals(moleculeFormulae))
 				{
-					this.formulaUpdateSB.append(prefix).append(" Old Formula: ").append(moleculeFormulae).append(" ; ").append("New Formula: ").append(firstFormula).append("\n");
+					this.formulaUpdateSB.append(reportLinePrefix).append(" Old Formula: ").append(moleculeFormulae).append(" ; ")
+														.append("New Formula: ").append(firstFormula).append("\n");
+					updated = true;
 				}
-				molecule.setAttributeValue(ReactomeJavaConstants.formula, firstFormula);
-				if (!testMode)
+				//else
+				//{
+					// Not updated.
+				//}
+			}
+			else
+			{
+				// Only print a warning if we're going from non-NULL formula to NULL formula.
+				if (moleculeFormulae != null && !moleculeFormulae.trim().equals(""))
 				{
-					adaptor.updateInstanceAttribute(molecule, ReactomeJavaConstants.formula);
+					logger.warn("Got empty/NULL formula for {}, old formula was: {}", molecule.toString(), moleculeFormulae);
 				}
```

Comment:
I can't remember -- did Peter confirm that it was okay to set the formula to null when we have a value in our database and ChEBI gives us an empty value?

Reply:
Peter had no comment. Lisa said:
> I think the formula additions/changes don't need to be reported.

And Bijay said:

> Information about formulae I don't think we use anywhere so those first 3 files can be done silently without curator involvement.

Reply:
Okay, sounds good!